### DISCUSSION TITLE
Flexible cookie storage

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#99] [Ideas]
The cookie storage is very limited. There is no way to set expiration time, `SameSite` attribute, etc.
The Crystal stdlib already has a `Cookie` class: https://crystal-lang.org/api/latest/HTTP/Cookie.html
Proposed API:
```crystal
cookies.encrypted << HTTP::Cookie.new(name: "user", value: "1", http_only: true)

# access
cookies.encrypted["user"].http_only
```

**Comment by ellmetha** [comment]
@notramo you can set cookie expiration times, `SameSite` policies, etc using the existing API already, by using the [`#set`](https://martenframework.com/docs/api/0.2/Marten/HTTP/Cookies.html#set%28name%3AString%7CSymbol%2Cvalue%2Cexpires%3ATime%7CNil%3Dnil%2Cpath%3AString%3D%22/%22%2Cdomain%3AString%7CNil%3Dnil%2Csecure%3ABool%3Dfalse%2Chttp_only%3ABool%3Dfalse%2Csame_site%3ANil%7CString%7CSymbol%3Dnil%29%3ANil-instance-method) method.

Though it is true that this method is not mentioned in the user documentation (except in the API docs) and I agree that the documentation around cookies could be enhanced.
